### ISSUE TITLE
Defeat the user inbox spaghetti monster

### ISSUE THREAD

**Issue by chmanie** [JoinColony/colonyDapp#1314]
![spaghetti monster](https://media.giphy.com/media/px6SrgxWZFJYY/giphy.gif)

## Description

The user inbox query grew into a big pile of spagehtti code. Plus it's not really working anymore. To reproduce on master:

Create a colony from scratch. You'll see the ENS entry in your inbox. Refresh the page, you'll see that the entry is gone completely, alongside with this error message in the console:

```
effects.js:52 Error: invalid hash - 0
    at checkHash (provider.js:92)
    at checkTopics (provider.js:349)
    at provider.js:345
    at Array.forEach (<anonymous>)
    at checkTopics (provider.js:344)
    at provider.js:345
    at Array.forEach (<anonymous>)
    at checkTopics (provider.js:344)
    at Object.topics (provider.js:65)
    at check (provider.js:51)
```

A few other things to consider:

* In `setupUserContext` we're calling this query for every colony a user is subscribed to. But if a user doesn't have any subscriptions, they could still have notifications (mentions for example). So what we need to do there is to just pass the list of subscribed colonies to the query (and get these in the saga to fetch the notifications and not in setupUserContext!)
* The query should also be called when opening the inbox not only once on dapp start (but see also the next bullet point)
* We're introducing store reactivity soon and the inbox query is definitely a very valid candidate to benefit from it. The refactor has to take that into consideration

## Also TODO
- [ ] Deal with the `putNotification` actions that are not surviving a reload (we probably want to get the native events for most of these events) -> open another issue if necessary.
